src/server/api/routes/nft/metadata.js

```diff
@@ -0,0 +1,165 @@
+      getNFTClassAndISCNMetadata(classId),
+      axios.get(
+        `${LIKECOIN_CHAIN_API}/likechain/likenft/v1/owner?class_id=${classId}`
+      ),
+      axios.get(
+        `${LIKECOIN_CHAIN_API}/likechain/likenft/v1/listings/${classId}`
+      ),
+      axios
+        .get(`${LIKECOIN_API_BASE}/likernft/purchase?class_id=${classId}`)
+        .catch(err => {
+          // skip 404 error for non Writing NFT
+          if (err.response?.status === 404) {
+            return null;
+          }
+          throw err;
+        }),
+    ]);
+
+    const owners = ownerInfoRes.data?.owners || [];
+    const listingInput = listingInfoRes.data?.listing || [];
+    const listings = formatAndFilterListing(listingInput, owners);
+    const purchaseInfo = purchaseInfoRes?.data;
+
+    res.json({
+      classData,
+      iscnData,
+      owners,
+      listings,
+      purchaseInfo,
+    });
+  } catch (err) {
+    if (err.message === 'NFT_CLASS_NOT_FOUND') {
+      res.status(404).send(err.message);
+    } else {
+      handleRestfulError(req, res, next, err);
+    }
+  }
+});
+
+async function getNFTClassAndISCNMetadata(classId) {
+  const {
+    data: { class: chainMetadata },
+  } = await axios
+    .get(`${LIKECOIN_CHAIN_API}/cosmos/nft/v1beta1/classes/${classId}`)
+    .catch(err => {
+      if (err.response?.data?.code === 2) {
+        // eslint-disable-next-line no-console
+        throw new Error('NFT_CLASS_NOT_FOUND');
+      }
+      throw err;
+    });
+  const {
+    name,
+    description,
+    uri,
+    data: { metadata, parent },
+  } = chainMetadata;
+  let classData = {
+    name,
+    description,
+    uri,
+    ...metadata,
+    parent,
+  };
+
+  if (isValidHttpUrl(uri)) {
+    const apiMetadataRes = await axios.get(uri).catch(err => {
+      if (err.response?.status !== 404) {
+        // eslint-disable-next-line no-console
+        console.error(err);
+      }
+    });
+    const apiMetadata = apiMetadataRes?.data;
+    if (apiMetadata && typeof apiMetadata === 'object') {
+      classData = { ...classData, ...apiMetadata };
+    }
+  }
+
+  let iscnRecord;
+  let iscnData;
+  const iscnId = parent?.iscn_id_prefix;
+  if (iscnId) {
+    const { data } = await axios
+      .get(`${LIKECOIN_CHAIN_API}/iscn/records/id?iscn_id=${iscnId}`)
+      .catch(err => {
+        if (!err.response?.status === 404) {
+          // eslint-disable-next-line no-console
+          console.error(err);
+        }
+      });
+    iscnRecord = data;
+    if (iscnRecord) {
+      iscnData = iscnRecord?.records?.[0]?.data;
+      classData.iscn_record = iscnData;
+    }
+  }
+
+  // in case nft chain metadata was not properly parsed
+  if (!(classData.iscn_owner || classData.account_owner)) {
+    if (iscnRecord) {
+      classData.iscn_owner = iscnRecord.owner;
+      classData.iscn_record_timestamp =
+        iscnRecord?.records?.[0]?.recordTimestamp;
+    } else if (parent.account) {
+      classData.account_owner = parent.account;
+    }
+  }
```

Comment:
We can always use `iscnRecord` here and ignore `classData.iscn_owner` I suppose

Reply:
Actually curious about why do we cast this schema inside NFT metadata but not access from ISCN metadata store?
Seems very customized?

Reply:
We didnt have to call iscn record when we first started, so putting iscn owner into api saves a call
But now we always call iscn api, so we should use data in iscn api instead

Reply:
Like the way but am afraid of introducing breaking change here 🌚
Try not to break sth ...

Reply:
The data from iscn is always correct, so as long as we use same key it is not breaking

Reply:
I meant remove `if (!(classData.iscn_owner || classData.account_owner)) {` and always assign `classData.iscn_owner = iscnRecord.owner;` ...